### DISCUSSION TITLE
Error when deploy hook is called

### DISCUSSION THREAD

**Discussion by zadama** [vercel/community#924] [Help]
Hey!

I'm getting the following error when my deploy hook is called:

error: RPC failed; HTTP 500 curl 22 The requested URL returned error: 500\n' + 'fatal: the remote end hung up unexpectedly\n'

This occurs during the cloning of the repository step of the execution. It hangs up for a few minutes then throws the error printed previously.

**Comment by amyegan** [comment]
Sorry you ran into a problem @zadama. 😞 Is the error still happening? This could have simply been a temporary connection issue with the remote repository. 

Please share more information you can about your project, such as 
- which remote git host you are using (GitHub, GitLab, etc)
- an outline of your deployment workflow
- any other warnings or errors that appear
- how long things hang before the 500 error appears
- anything else you can think of that might help with investigation

**Comment by zadama** [comment]
Yes it is unfortunately. It's been happening for about 1-2 weeks actually. 

- I'm using Bitbucket
- I have a deploy hook set on my production branch, master. It has worked great for the past few months but just recently for the past two weeks it has stopped working. I should also mention that it's a Next.js project, if it makes any difference.
- No, the warning posted is the only one that appears. At the end of the warning/error it prints "Error: cloning failed."
- It hangs for about 8-10 minutes before the error appears.

**Reply by amyegan** [reply]
Thank you for sharing this information. I know this is a frustrating problem to face, so I appreciate your kindness and patience. I'm sending it the build engineers and will update here when we know more about the source of the error.

I don't know if you have already found another way to deploy your project, and I don't want to leave you without other options. If you need a temporary workaround to deploy a change to your project, the [CLI](https://vercel.com/docs/cli) might work for you to manually deploy updates from your local computer.

**Reply by amyegan** [reply]
Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public) to help the team investigate this problem? They have not been able to replicate the error yet, so it seems to require a specific setting or configuration

**Reply by zadama** [reply]
> Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public) to help the team investigate this problem? They have not been able to replicate the error yet, so it seems to require a specific setting or configuration

I was able to solve it temporarily by making a new commit to my master/deploy branch. That solution, however, seemed to only work for a day or two, now I'm back with the same error.

**Reply by amyegan** [reply]
Are you able to reproduce this problem with another project/repository? If it only happens in your main project repo, is that something you would be able to share with us so we can see the problem as it occurs? 

Even if you  can't share any code, please share examples of successful and unsuccessful builds. You can share the link from your dashboard, the deployment ID, or the unique deployment domain. Any of those will help me locate more specific information in the logs.
